src/qt/receivecoinsdialog.cpp

```diff
@@ -92,10 +93,16 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
         // Last 2 columns are set by the columnResizingFixer, when the table geometry is ready.
         columnResizingFixer = new GUIUtil::TableViewLastColumnResizingFixer(tableView, AMOUNT_MINIMUM_COLUMN_WIDTH, DATE_COLUMN_WIDTH, this);
 
-        if (model->wallet().getDefaultAddressType() == OutputType::BECH32) {
-            ui->useLegacyAddress->setCheckState(Qt::Unchecked);
+        if (model->node().isAddressTypeSet()) {
```

Comment:
nit, to avoid nesting previous code you could do:
```cpp
if (!model->node().isAddressTypeSet()) {
    // Always fall back to bech32 in the gui
    ui->useLegacyAddress->setCheckState(Qt::Unchecked);
} else if ...

Reply:
Good point. I will leave it as is for now, to not invalidate review.